types/evidence.go

```diff
@@ -172,6 +174,20 @@ func (dve *DuplicateVoteEvidence) Equal(ev Evidence) bool {
 	return bytes.Equal(dveHash, evHash)
 }
 
+// ValidateBasic performs basic validation.
+func (dve *DuplicateVoteEvidence) ValidateBasic() error {
+	if len(dve.PubKey.Bytes()) == 0 {
+		return errors.New("Empty PubKey")
+	}
+	if err := dve.VoteA.ValidateBasic(); err != nil {
```

Comment:
These votes are pointers. We either need to make them non-pointers, or check that they're not nil before dereferencing to validate.

Reply:
Awesome find

Reply:
Thank @Slamper . He's killing it on the bug bounty

Reply:
Thanks @ebuchman :+1:  Always happy to contribute!